routes/payments/blockonomics.js

```diff
@@ -0,0 +1,149 @@
+            };
+            
+            // send the email with the response
+            // TODO: Should fix this to properly handle result            
+            common.sendEmail(req.session.paymentEmailAddr, 'Your payment with ' + config.cartTitle, common.getEmailTemplate(paymentResults));
+            res.status(200).json({ err: '' });        
+                        
+        }catch(ex){
+            console.info('Error updating status success blockonomics', ex);
+            res.status(200).json({ err: 'Error updating status' });        
+
+        }
+        return;
+      }
+      console.info('Amount not sufficient blockonomics', address);
+      try{
+          await db.orders.updateOne({
+              _id: order._id },
+              { $set: { orderStatus: 'Declined', orderReceivedBtc: amount }
+          }, { multi: false });
+      }catch(ex){
+          console.info('Error updating status insufficient blockonomics', ex);
+      }
+      res.status(200).json({ err: 'Amount not sufficient' });             
+      return;
+    }
+    res.status(200).json({ err: 'Order not found' });             
+    console.info('Order not found blockonomics', address);
+    return;
+  }
+  res.status(200).json({ err: 'Payment not final' });             
+  console.info('Payment not final blockonomics', address);
+
+});
+
+router.post('/checkout_action', (req, res, next) => {
+
+  const blockonomicsConfig = common.getPaymentConfig();
+  const config = req.app.config;
+  const db = req.app.db;
+  var blockonomicsParams = {};
+  // get current rate
+  unirest
+  .get(blockonomicsConfig.hostUrl+blockonomicsConfig.priceApi+config.currencyISO)
+  .then((response) => {
+    blockonomicsParams.expectedBtc = Math.round(req.session.totalCartAmount / response.body.price * Math.pow(10, 8)) / Math.pow(10, 8);
+    // get new address
+    unirest
+      .post(blockonomicsConfig.hostUrl+blockonomicsConfig.newAddressApi)
+      .headers({'Content-Type': 'application/json', 'User-Agent': 'blockonomics','Accept': 'application/json', 'Authorization': 'Bearer ' + blockonomicsConfig.apiKey})
+      .then((response) => {
+          blockonomicsParams.address = response.body.address;
+          blockonomicsParams.timestamp = Math.floor(new Date() / 1000);
+          // create order with status Pending and save ref
+        
+            const orderDoc = {
+                orderPaymentId: blockonomicsParams.address,
+                orderPaymentGateway: 'Blockonomics',
+                orderExpectedBtc: blockonomicsParams.expectedBtc,
+                orderTotal: req.session.totalCartAmount,
+                orderShipping: req.session.totalCartShipping,
+                orderItemCount: req.session.totalCartItems,
+                orderProductCount: req.session.totalCartProducts,
+                orderEmail: req.session.customerEmail,

```

Comment:
Please add the new field using the following line below line 105:

`
orderCompany: req.session.customerCompany,
`

Reply:
What's order company? Which new field?